src_c/math.c

```diff
@@ -4233,6 +4245,87 @@ math_clamp(PyObject *self, PyObject *const *args, Py_ssize_t nargs)
     return value;
 }
 
+#define RAISE_ARG_TYPE_ERROR(py_num_obj, var)                         \
+    if (!PyNumber_Check(py_num_obj)) {                                \
+        return RAISE(PyExc_TypeError,                                 \
+                     "The argument '" var "' must be a real number"); \
+    }
+
+static PyObject *
+math_invlerp(PyObject *self, PyObject *const *args, Py_ssize_t nargs)
+{
+    if (nargs != 3)
+        return RAISE(PyExc_TypeError,
+                     "invlerp requires exactly 3 numeric arguments");
+
+    PyObject *min = args[0];
+    RAISE_ARG_TYPE_ERROR(min, "min")
+    PyObject *max = args[1];
+    RAISE_ARG_TYPE_ERROR(max, "max")
+    PyObject *value = args[2];
+    RAISE_ARG_TYPE_ERROR(value, "value")
+
+    // if (!PyNumber_Check(min) || !PyNumber_Check(max) ||
+    // !PyNumber_Check(value))
+    //     return RAISE(PyExc_TypeError,
+    //                  "invlerp requires all the arguments to be numbers");
+
+    double t = PyFloat_AsDouble(value);
+    double a = PyFloat_AsDouble(min);
+    double b = PyFloat_AsDouble(max);
```

Comment:
```suggestion
    double a = PyFloat_AsDouble(args[0]);
    RAISE_ARG_TYPE_ERROR("min")
    double b = PyFloat_AsDouble(args[1])
    RAISE_ARG_TYPE_ERROR("max")
    double t = PyFloat_AsDouble(args[2]);
    RAISE_ARG_TYPE_ERROR("value")
```
We check if `PyFloat_AsDouble` raises an error, and if it does, it means that the input is incorrect and therefore raise our custom error that tells which argument exactly is incorrect.

Sorry for the back and forth about the reviews, but if this is resolved I don't think I will have any other issues with this PR. You can also apply a similar patch for the other function being added